client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/Cache.java

```diff
@@ -61,6 +61,19 @@
      */
     Future<V> put(K key, V value, long lifespan, TimeUnit lifespanUnit);
 
+    /**
+     * Replaces the entry for a key only if currently mapped to a given value.
+     *
+     * @param key The key.
+     * @param oldValue The value to overwrite.
+     * @param newValue The value to store.
+     * @param lifespan The lifespan of the entry. A negative value is interpreted as an unlimited lifespan.
+     * @param lifespanUnit The time unit for the lifespan.
+     * @return A succeeded future containing a boolean, indicating whether the value was replaced or not.
+     *         A failed future if the value could not be stored in the cache.
+     */
+    Future<Boolean> replace(K key, V oldValue, V newValue, long lifespan, TimeUnit lifespanUnit);
```

Comment:
what if any of the params are `null`?

Reply:
I've added checks and adapted the javadoc (also for the other methods).